resources/us_verifications/models/components.yml

```diff
@@ -0,0 +1,232 @@
+type: object
+
+required:
+  - primary_number
+  - street_predirection
+  - street_name
+  - street_suffix
+  - street_postdirection
+  - secondary_designator
+  - secondary_number
+  - pmb_designator
+  - pmb_number
+  - extra_secondary_designator
+  - extra_secondary_number
+  - city
+  - state
+  - zip_code
+  - zip_code_plus_4
+  - zip_code_type
+  - delivery_point_barcode
+  - address_type
+  - record_type
+  - default_building_address
+  - county
+  - county_fips
+  - carrier_route
+  - carrier_route_type
+  - latitude
+  - longitude
+
+properties:
+  primary_number:
+    type: string
+    description: >
+      The numeric or alphanumeric part of an address preceding the
+      street name. Often the house, building, or PO Box number.
+    example: "185"
+    readOnly: true
+
+  street_predirection:
+    type: string
+    description: >
+      Geographic direction preceding a street name (<code>N</code>, <code>S</code>, <code>E</code>,
+      <code>W</code>, <code>NE</code>, <code>SW</code>, <code>SE</code>, <code>NW</code>).
+    example: ""
+    readOnly: true
+
+  street_name:
+    type: string
+    description: >
+      The name of the street.
+    example: BERRY
+    readOnly: true
+
+  street_suffix:
+    type: string
+    description: >
+      The standard USPS abbreviation for the street suffix (<code>ST</code>, <code>AVE</code>, <code>BLVD</code>, etc).
+    example: ST
+    readOnly: true
+
+  street_postdirection:
+    type: string
+    description: >
+      Geographic direction following a street name (<code>N</code>, <code>S</code>, <code>E</code>,
+      <code>W</code>, <code>NE</code>, <code>SW</code>, <code>SE</code>, <code>NW</code>).
+    example: ""
+    readOnly: true
+
+  secondary_designator:
+    type: string
+    description: >
+      The standard USPS abbreviation describing the <code>components[secondary_number]</code>
+      (<code>STE</code>, <code>APT</code>, <code>BLDG</code>, etc).
+    example: STE
+    readOnly: true
+
+  secondary_number:
+    type: string
+    description: >
+      Number of the apartment/unit/etc.
+    example: "6100"
+    readOnly: true
+
+  pmb_designator:
+    type: string
+    description: >
+      Designator of a [CMRA-authorized](https://en.wikipedia.org/wiki/Commercial_mail_receiving_agency) private mailbox.
+    example: ""
+    readOnly: true
+
+  pmb_number:
+    type: string
+    description: >
+      Number of a [CMRA-authorized](https://en.wikipedia.org/wiki/Commercial_mail_receiving_agency) private mailbox.
+    example: ""
+    readOnly: true
+
+  extra_secondary_designator:
+    type: string
+    description: >
+      An extra (often unnecessary) secondary designator provided with the input address.
+    example: ""
+    readOnly: true
+
+  extra_secondary_number:
+    type: string
+    description: >
+      An extra (often unnecessary) secondary number provided with the input address.
+    example: ""
+    readOnly: true
+
+  city:
+    readOnly: true
+    allOf:
```

Comment:
So even though the linters don't flag these, they aren't conforming to the standard:

" allOf takes an array of object definitions that are validated independently but together compose a single object."

Go ahead and leave out the `ReadOnly`s, as the behavior of those change in v3.1.